src/database-controller/src/core/framework.js

```diff
@@ -0,0 +1,421 @@
+      this.setGeneration(1);
+    }
+    return parseInt(this._snapshot.metadata.annotations.requestGeneration);
+  }
+}
+
+// Class Add-ons handles creation/patching/deletion of job add-ons.
+// Currently there are 3 types of add-ons: configSecret, priorityClass, and dockerSecret.
+class AddOns {
+  constructor(
+    configSecretDef = null,
+    priorityClassDef = null,
+    dockerSecretDef = null,
+  ) {
+    if (configSecretDef !== null && !(configSecretDef instanceof Object)) {
+      this._configSecretDef = JSON.parse(configSecretDef);
+    } else {
+      this._configSecretDef = configSecretDef;
+    }
+    if (priorityClassDef !== null && !(priorityClassDef instanceof Object)) {
+      this._priorityClassDef = JSON.parse(priorityClassDef);
+    } else {
+      this._priorityClassDef = priorityClassDef;
+    }
+    if (dockerSecretDef !== null && !(dockerSecretDef instanceof Object)) {
+      this._dockerSecretDef = JSON.parse(dockerSecretDef);
+    } else {
+      this._dockerSecretDef = dockerSecretDef;
+    }
+  }
+
+  async create() {
+    if (this._configSecretDef) {
+      try {
+        await k8s.createSecret(this._configSecretDef);
+      } catch (err) {
+        if (err.response && err.response.statusCode === 409) {
+          logger.warn(
+            `Secret ${this._configSecretDef.metadata.name} already exists.`,
+          );
+        } else {
+          throw err;
+        }
+      }
+    }
+    if (this._priorityClassDef) {
+      try {
+        await k8s.createPriorityClass(this._priorityClassDef);
+      } catch (err) {
+        if (err.response && err.response.statusCode === 409) {
+          logger.warn(
+            `PriorityClass ${this._priorityClassDef.metadata.name} already exists.`,
+          );
+        } else {
+          throw err;
+        }
+      }
+    }
+    if (this._dockerSecretDef) {
+      try {
+        await k8s.createSecret(this._dockerSecretDef);
+      } catch (err) {
+        if (err.response && err.response.statusCode === 409) {
+          logger.warn(
+            `Secret ${this._dockerSecretDef.metadata.name} already exists.`,
+          );
+        } else {
+          throw err;
+        }
+      }
+    }
+  }
+
+  silentPatch(frameworkResponse) {
+    // do not await for patch
+    this._configSecretDef &&
+      k8s
+        .patchSecretOwnerToFramework(this._configSecretDef, frameworkResponse)
+        .catch(ignoreError);
+    this._dockerSecretDef &&
+      k8s
+        .patchSecretOwnerToFramework(this._dockerSecretDef, frameworkResponse)
+        .catch(ignoreError);
+  }
+
+  silentDelete() {
+    // do not await for delete
+    this._configSecretDef &&
+      k8s.deleteSecret(this._configSecretDef.metadata.name).catch(ignoreError);
+    this._priorityClassDef &&
+      k8s
+        .deletePriorityClass(this._priorityClassDef.metadata.name)
+        .catch(ignoreError);
+    this._dockerSecretDef &&
+      k8s.deleteSecret(this._dockerSecretDef.metadata.name).catch(ignoreError);
+  }
```

Comment:
At least log error

Reply:
Actually `ignoreError` does the logging thing.

Reply:
I change the function name `ignoreError` to `logError` to make it more clear.